src/datachain/sql/sqlite/types.py

```diff
@@ -28,26 +28,21 @@ def get_col_spec(self, **kwargs):
 
 
 def adapt_array(arr):
-    return orjson.dumps(arr).decode("utf-8")
+    return json.dumps(arr, ensure_ascii=False, separators=(",", ":"))
 
 
 def adapt_dict(dct):
-    return orjson.dumps(dct).decode("utf-8")
+    return json.dumps(dct, ensure_ascii=False, separators=(",", ":"))
 
 
 def convert_array(arr):
-    return orjson.loads(arr)
+    return json.loads(arr)
 
 
 def adapt_np_array(arr):
-    def _json_serialize(obj):
-        if isinstance(obj, np.ndarray):
-            return obj.tolist()
-        return obj
-
-    return orjson.dumps(
-        arr, option=orjson.OPT_SERIALIZE_NUMPY, default=_json_serialize
-    ).decode("utf-8")
+    # Primarily needed for UDF numpy results (e.g. WDS)
+    # tolist() gives nested Python lists + native scalars; json.dumps handles NaN/Inf.
+    return json.dumps(arr.tolist(), ensure_ascii=False, separators=(",", ":"))
```

Comment:
@dreadatour this bother me the most? what the impact of ujson here on WDS? also cc @dmpetrov ...

we can still keep the previous implementation ... but it will be breaking on NaN / Inf (which can be happening in scientific results) or we will have to handle them somehow which might slow down ...

Reply:
`ujson` was two times faster: https://github.com/iterative/dvcx/pull/1098

Basic test:

```python
In [1]: import math, ujson

In [2]: ujson.dumps(float("nan"))
Out[2]: 'NaN'

In [3]: math.isnan(ujson.loads(ujson.dumps(float("nan"))))
Out[3]: True

In [4]: ujson.dumps(float("inf"))
Out[4]: 'Infinity'

In [5]: math.isinf(ujson.loads(ujson.dumps(float("inf"))))
Out[5]: True

In [6]: ujson.dumps(float("-inf"))
Out[6]: '-Infinity'

In [7]: math.isinf(ujson.loads(ujson.dumps(float("-inf"))))
Out[7]: True

In [8]: ujson.dumps([float("nan"), float("inf"), float("-inf"), 37])
Out[8]: '[NaN,Infinity,-Infinity,37]'

In [9]: ujson.loads(ujson.dumps([float("nan"), float("inf"), float("-inf"), 37]))
Out[9]: [nan, inf, -inf, 37]
```